### PR TITLE
use cantabular api extension: GetDimensionsByName

### PR DESCRIPTION
**Pull request by DavidSubiros**
### What

- replaced `GetCodebook` with `GetDimensionsByName`, which does a graphQL call against Cantabular API Extension
- added new config paramter for Cantabular API Ext URL
- added healthcheck checker for Cantabular API Ext
- fixed unit and component tests accordingly

Using:
- new version of dp-api-clients-go: https://github.com/ONSdigital/dp-api-clients-go/pull/166
- new version of dp-kafka: https://github.com/ONSdigital/dp-kafka/pull/52

### How to review

- Make sure code changes make sense
- Make sure unit/component tests pass
- (info) tested locally with dp-compose and full import/export script

note: requires the following dp-componse to be checked out in order to test locally:
https://github.com/ONSdigital/dp-compose/pull/66

### Who can review

anyone

### CODE REVIEW

**Review by redhug1 — CHANGES_REQUESTED** [review]
Please can you fix these:
service/service.go
lines 71, 100, 209 : Unhandled error

cmd/test-consumer/main.go
lines 105, 108 : Unhandled error

features/steps/component.go
lines 114, 147, 200, 210, 223 : Unhandled error
lines 30, 31, 32 : Type can be omitted

**Review by redhug1 — APPROVED** [review]
LGTM

**Review by redhug1 — CHANGES_REQUESTED** [review]
I think you meant to update kafka lib to v3.1.1

File: ci/scripts/component.sh [hunk 1]
```diff
@@ -9,4 +9,4 @@ popd
 exit $e
 
 # Show message to prevent any confusion by 'ERROR 0' outpout
-echo "please ignore error codes 0, like so: ERRO[xxxx] 0, as error code 0 means that there was no error"
\ No newline at end of file
+echo "please ignore error codes 0, like so: ERROR[xxxx] 0, as error code 0 means that there was no error"
```
Comment (redhug1) [inline ci/scripts/component.sh:12]: needs a CR here

**Review by redhug1 — CHANGES_REQUESTED** [review]
kafka update to v3.1.1 in go mod ...

plus other single comment.

**Comment by DavidSubiros** [comment]
> I think you meant to update kafka lib to v3.1.1

yes, it should be updated in this commit: https://github.com/ONSdigital/dp-import-cantabular-dataset/pull/37/commits/cb49b6403e94080ed83053c1d6e30a9a53e45964

**Review by redhug1 — CHANGES_REQUESTED** [review]
Those changes look OK, but:

I thought i had already reported these problems with dp-import-cantabular-dataset in github, but i could be wrong, so here they are again, just in case:

component.go line 118 is currently:
	if c.consumer.Start(); err != nil {
which is showing an unhandled error, so i think maybe it should be:
	if err = c.consumer.Start(); err != nil {

A similar error exists in:

 main.go, line 110

Also, line 223 in service.go is an Unhandled error

**Review by redhug1 — APPROVED** [review]
LGTM

**Pull request merged**